test/smoke/webhookoncall_test.go

```diff
@@ -0,0 +1,84 @@
+package smoke
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/target/goalert/expflag"
+	"github.com/target/goalert/test/smoke/harness"
+)
+
+type POSTDataOnCallUser struct {
+	ID   string
+	Name string
+	URL  string
+}
+
+type POSTDataOnCallNotification struct {
+	AppName      string
+	Type         string
+	Users        []POSTDataOnCallUser
+	ScheduleID   string
+	ScheduleName string
+	ScheduleURL  string
+}
+
+// TestWebhookOnCallNotification tests that the configured rule sends the intended notification to the webhook.
+func TestWebhookOnCallNotification(t *testing.T) {
+	t.Parallel()
+
+	ch := make(chan POSTDataOnCallNotification)
```

Comment:
When setting up a receiver like this, add a buffer if possible. In this case, it will alow the handler goroutine to exit when it's done, even if the test were to exit early or something. (otherwise it will block forever on `ch <- alert`)
```suggestion
	ch := make(chan POSTDataOnCallNotification, 1)
```

Reply:
this would apply to the other webhook test as well right? https://github.com/target/goalert/blob/05efe4a36d0e7ce87e0b1c825cd170d87211ef68/test/smoke/webhook_test.go#L25C25-L25C25

Reply:
yeah good catch!